cores/esp32/Arduino.h

```diff
@@ -127,7 +127,7 @@ void loop(void);
 
 long random(long, long);
 void randomSeed(unsigned long);
-long map(long, long, long, long, long);
+long Arduino_map(long, long, long, long, long);
```

Comment:
I'm not sure if this is a good idea at all. Maybe undefine the other map? Arduino users expect this to work this way, so any sketch that uses it will break ;)

Reply:
The other map is the regular C++ libstdc++ map which is being used in all C++ books, C++ projects, etc.. C++ is now here with this for more than 20 years. It is a fault by Arduino to use this name. It is also little being used compared to the C++ map.

Reply:
I understand, but this is Arduino framework so it should adhere to the well known Arduino API ;) we can not rename or get rid of this all together.

Reply:
maybe it can be renamed to Arduino_map and then define map to Arduino_map (which could in turn be later undefined to allow for C++ map)

Reply:
Arduino uses the gcc compiler and C runtime which brings all the standard runtime libraries. This goes first and than the Arduino libs are on top of it.

Reply:
sure, but you tell me with this change how will all sketches and examples out there that use Arduino's "map" work?